### PR TITLE
Remove the shell completion command from the CLI

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
- Use the program name in help message instead of the project name
- Remove the shell completion command from the CLI tool and add a new separate go script to generate autocompletion scripts
- Add the shell completion script(mi_bash_completion.sh) to the Linux zip archive
- Update the README

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: cmd/README.md [hunk 1]
```diff
@@ -20,6 +20,9 @@ Command Line tool for managing the WSO2 Micro Integrator
 
     Created Command Line tool packages will be available at cmd/build directory
 
+- #### Generating Autocompletion scripts
+    After changing commands run following to generate shell completions `go run tools/completion.go`
```
Comment (a5anka) [inline cmd/README.md]: I did not understand this. Can you explain what do you mean by "After changing commands "?

File: cmd/README.md [hunk 2]
```diff
@@ -29,8 +32,8 @@ Command Line tool for managing the WSO2 Micro Integrator
 
     NOTE: To execute the tool from anywhere, append the location of the executable (mi) to your $PATH variable.
 
-- ### Command Autocompletion
-    Copy the file mi_bash_completion.sh to /etc/bash_completion.d/ and source it with source /etc/bash_completion.d/mi_bash_completion.sh to enable bash auto-completion.
+- ### Command Autocompletion (For Bash Only)
+    Copy the file `mi_bash_completion.sh` in `shell-completions` directory to `/etc/bash_completion.d/` and source it with source `/etc/bash_completion.d/mi_bash_completion.sh` to enable bash auto-completion.
```
Comment (a5anka) [inline cmd/README.md]:
```suggestion
    Copy the file `mi_bash_completion.sh` in `shell-completions` directory to `/etc/bash_completion.d/` and source it with `source /etc/bash_completion.d/mi_bash_completion.sh` to enable bash auto-completion.
```

File: cmd/README.md (hunk 1 above)
Comment (a5anka) [reply cmd/README.md]: Isn't it better if we avoid this step and just have the mi_bash_completion.sh as a resource that we can download?

Comment (SanojPunchihewa) [reply cmd/README.md]: If you add new commands or change an existing command, to get the new shell completion script you must run the above command

File: cmd/README.md (hunk 2 above)
Comment (SanojPunchihewa) [reply cmd/README.md]: Fixed in dce7fcd

File: cmd/README.md (hunk 1 above)
Comment (SanojPunchihewa) [reply cmd/README.md]: mi_bash_completion.sh script is added to the zip archive in 157ab87

**Comment by SanojPunchihewa** [comment]
@menuka94 @a5anka WDYT about 40bbcdd?

File: cmd/utils/utils.go [hunk 3]
```diff
@@ -169,3 +169,13 @@ func GetCmdUsage(program, cmd, subcmd, arg string) string {
 		"  " + program + " " + cmd + " " + subcmd + "(s) " + arg + "\n\n"
 	return showCmdUsage
 }
+
+func GetProgramName() string {
+	var programName = os.Args[0]
+	// Replace the dynamic program name with the poject name(mi) to generate bash completion scripts
+	// when the user use go run tools/completeion.go
+	if strings.Contains(programName, "completion") {
```
Comment (a5anka) [inline cmd/utils/utils.go]: This not very readable and not good to have a completion specific logic in runtime code.

Comment (menuka94) [reply cmd/utils/utils.go]: But we need to make use of Cobra's code completion functionality as well. Dynamically retrieving the program name from `os.Args[0]` does not work in the case of 'go build

**Comment by menuka94** [comment]
> @menuka94 @a5anka WDYT about [40bbcdd](https://github.com/wso2/micro-integrator/commit/40bbcddac21d2e884946df87b482036e95354915)?

We should somehow make use of Cobra's "generate auto-completion scripts" feature as it is important in terms of user experience.

**Comment by a5anka** [comment]
> > @menuka94 @a5anka WDYT about [40bbcdd](https://github.com/wso2/micro-integrator/commit/40bbcddac21d2e884946df87b482036e95354915)?
> 
> We should somehow make use of Cobra's "generate auto-completion scripts" feature as it is important in terms of user experience.

I agree. What I say is that this is not the way.

File: cmd/utils/utils.go (hunk 3 above)
Comment (SanojPunchihewa) [reply cmd/utils/utils.go]: Fixed in 80277ce

File: cmd/build.sh [hunk 4]
```diff
@@ -136,6 +136,13 @@ else
     echo "Skipping Go Tests..."
 fi
 
+# run the completion.go file to get the bash completion script
+# To do the string replace first build the script so that we have a consistent name
+go build -gcflags=-trimpath=$GOPATH -asmflags=-trimpath=$GOPATH tools/completion.go
+./completion
```
Comment (a5anka) [inline cmd/build.sh]: Shall we use a longer placeholder word instead of ./completion to avoid mistakes?

Comment (SanojPunchihewa) [reply cmd/build.sh]: Fixed in e513c8c

**Comment by a5anka** [comment]
> Shall we include the bash auto-completion script for Mac OS packaging as well?

@menuka94 @SanojPunchihewa Did we do this?

**Comment by SanojPunchihewa** [comment]
> > Shall we include the bash auto-completion script for Mac OS packaging as well?
> 
> @menuka94 @SanojPunchihewa Did we do this?

Yes it was added in b8baad2

**Pull request merged**